include/fastscapelib/profile_grid.hpp

```diff
@@ -0,0 +1,283 @@
+    class profile_grid_xt : public structured_grid<profile_grid_xt<XT>>
+    {
+    public:
+
+        using self_type = profile_grid_xt<XT>;
+        using base_type = structured_grid<self_type>;
+        using inner_types = grid_inner_types<self_type>;
+
+        using xt_selector = typename inner_types::xt_selector;
+        using xt_ndims = typename inner_types::xt_ndims;
+
+        using size_type = typename inner_types::size_type;
+        using shape_type = typename inner_types::shape_type;
+
+        using spacing_type = typename inner_types::spacing_type;
+        using neighbors_type = typename inner_types::neighbors_type;
+
+        using boundary_status_type = typename inner_types::boundary_status_type;
+        using node_status_type = typename inner_types::node_status_type;
+
+        profile_grid_xt(size_type size,
+                        spacing_type spacing,
+                        const boundary_status_type& status_at_bounds,
+                        const std::vector<node>& status_at_nodes = {});
+
+    protected:
+
+        shape_type m_shape;
+        size_type m_size;
+        spacing_type m_spacing;
+
+        node_status_type m_status_at_nodes;
+        boundary_status_type m_status_at_bounds;
+        void set_status_at_nodes(const std::vector<node>& status_at_nodes);
+
+        static constexpr std::array<std::ptrdiff_t, 3> offsets { {0, -1, 1} };
+        std::vector<neighbors_type> m_all_neighbors;
+        void precompute_neighbors();
+
+        const neighbors_type& neighbors_impl(std::size_t idx) const;
+
+        friend class structured_grid<self_type>;
+    };
+
+    template <class XT>
+    constexpr std::array<std::ptrdiff_t, 3> profile_grid_xt<XT>::offsets;
+
+    /**
+     * @name Constructors
+     */
+    //@{
+    /**
+     * Creates a new profile grid.
+     *
+     * @param size Total number of grid nodes.
+     * @param spacing Distance between two adjacent grid nodes.
+     * @param status_at_bounds Status at boundary nodes (left & right grid edges).
+     * @param status_at_nodes Manually define the status at any node on the grid.
+     */
+    template <class XT>
+    profile_grid_xt<XT>::profile_grid_xt(size_type size,
+                                        spacing_type spacing,
+                                        const boundary_status_type& status_at_bounds,
+                                        const std::vector<node>& status_at_nodes)
+        : base_type(), m_size(size), m_spacing(spacing), m_status_at_bounds(status_at_bounds)
+    {
+        m_shape = {static_cast<typename shape_type::value_type>(m_size)};
+        set_status_at_nodes(status_at_nodes);
+        precompute_neighbors();
+    }
+    //@}
+
+    template <class XT>
+    void profile_grid_xt<XT>::set_status_at_nodes(const std::vector<node>& status_at_nodes)
+    {
+        node_status_type temp_status_at_nodes(m_shape, node_status::core);
+
+        temp_status_at_nodes(0) = m_status_at_bounds.left;
```

Comment:
We should also check that `status_at_bounds` does not contain `node_status::looped_boundary` (see my comment above).